ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/SnapWorldDownloadState.java

```diff
@@ -441,8 +437,7 @@ public BlockAddedObserver createBlockchainObserver() {
       final boolean isBlockchainCaughtUp =
           snapSyncState.isWaitingBlockchain() && !pivotBlockSelector.isBlockchainBehind();
 
-      if (isNewPivotBlockFound
-          || isBlockchainCaughtUp) { // restart heal if we found a new pivot block or if close to
+      if (snapSyncState.isHealTrieInProgress() && (isNewPivotBlockFound || isBlockchainCaughtUp)) {
         // head again
```

Comment:
the rest of this comment has been deleted - doesn't make sense by itself

Reply:
Done. Removed it, don't think we need the comment